internal/database/webhook_logs.go

```diff
@@ -0,0 +1,374 @@
+	}
+
+	return preds
+}
+
+func (s *webhookLogStore) Count(ctx context.Context, opts WebhookLogListOpts) (int64, error) {
+	q := sqlf.Sprintf(
+		webhookLogCountQueryFmtstr,
+		sqlf.Join(opts.predicates(), " AND "),
+	)
+
+	row := s.QueryRow(ctx, q)
+	var count int64
+	if err := row.Scan(&count); err != nil {
+		return 0, err
+	}
+
+	return count, nil
+}
+
+func (s *webhookLogStore) List(ctx context.Context, opts WebhookLogListOpts) ([]*types.WebhookLog, int64, error) {
+	preds := opts.predicates()
+	if cursor := opts.Cursor; cursor != 0 {
+		preds = append(preds, sqlf.Sprintf("id <= %s", cursor))
+	}
+
+	var limit *sqlf.Query
+	if opts.Limit != 0 {
+		limit = sqlf.Sprintf("LIMIT %s", opts.Limit+1)
+	} else {
+		limit = sqlf.Sprintf("")
+	}
+
+	q := sqlf.Sprintf(
+		webhookLogListQueryFmtstr,
+		sqlf.Join(webhookLogColumns, ", "),
+		sqlf.Join(preds, " AND "),
+		limit,
+	)
+
+	rows, err := s.Query(ctx, q)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer rows.Close()
+
+	logs := []*types.WebhookLog{}
+	for rows.Next() {
+		log := types.WebhookLog{}
+		if err := s.scanWebhookLog(ctx, &log, rows); err != nil {
+			return nil, 0, err
+		}
+		logs = append(logs, &log)
+	}
+
+	var next int64 = 0
+	if opts.Limit != 0 && len(logs) == opts.Limit+1 {
+		next = logs[len(logs)-1].ID
+		logs = logs[:len(logs)-1]
+	}
+
+	return logs, next, nil
+}
+
+func (s *webhookLogStore) DeleteStale(ctx context.Context, retention time.Duration) error {
+	before := time.Now().Add(-retention)
+
+	q := sqlf.Sprintf(
+		webhookLogDeleteStaleQueryFmtstr,
+		before,
+	)
+
+	return s.Exec(ctx, q)
+}
+
+var webhookLogColumns = []*sqlf.Query{
+	sqlf.Sprintf("id"),
+	sqlf.Sprintf("received_at"),
+	sqlf.Sprintf("external_service_id"),
+	sqlf.Sprintf("status_code"),
+	sqlf.Sprintf("request"),
+	sqlf.Sprintf("response"),
+	sqlf.Sprintf("encryption_key_id"),
+}
+
+const webhookLogCreateQueryFmtstr = `
+-- source: internal/database/webhook_logs.go:Create
+INSERT INTO
+	webhook_logs (
+		received_at,
+		external_service_id,
+		status_code,
+		request,
+		response,
+		encryption_key_id
+	)
+	VALUES (
+		%s,
+		%s,
+		%s,
+		%s,
+		%s,
+		%s
+	)
+	RETURNING %s
+`
+
+const webhookLogGetByIDQueryFmtstr = `
+-- source: internal/database/webhook_logs.go:GetByID
+SELECT
+	%s
+FROM
+	webhook_logs
+WHERE
+	id = %s
+`
+
+const webhookLogCountQueryFmtstr = `
+-- source: internal/database/webhook_logs.go:Count
+SELECT
+	COUNT(id)
+FROM
+	webhook_logs
+WHERE
+	%s
+`
+
+const webhookLogListQueryFmtstr = `
+-- source: internal/database/webhook_logs.go:List
+SELECT
+	%s
+FROM
+	webhook_logs
+WHERE
+	%s
+ORDER BY
+	id DESC
+%s -- LIMIT
+`
+
+const webhookLogDeleteStaleQueryFmtstr = `
+-- source: internal/database/webhook_logs.go:DeleteStale
+DELETE FROM
+	webhook_logs
+WHERE
+	received_at <= %s
+`
+
+func (s *webhookLogStore) scanWebhookLog(ctx context.Context, log *types.WebhookLog, sc interface {
+	Scan(...interface{}) error
+}) error {
```

Comment:
```suggestion
func (s *webhookLogStore) scanWebhookLog(ctx context.Context, log *types.WebhookLog, sc dbutil.Scanner) error {
```

I think this type exists

Reply:
That's what I get for copy pasting. Thanks!